test/test_cli.rb

```diff
@@ -147,6 +147,21 @@ def test_state
     assert_equal url, data["config"].options[:control_url]
   end
 
+  def test_state_file_callback_filtering
+    cli = Puma::CLI.new [ "--config", "test/config/state_file_testing_config.rb", 
+                          "--state", @tmp_path ]
+    cli.send( :parse_options )
```

Comment:
assuming this broken syntax is from the upstream PR?

Reply:
Yes, I just cherry-picked the commit.